mypy/checker.py

```diff
@@ -907,6 +907,14 @@ def check_override(self, override: FunctionLike, original: FunctionLike,
             # this could be unsafe with reverse operator methods.
             fail = True
 
+        if isinstance(original, CallableType) and isinstance(override, CallableType):
+            if (isinstance(original.definition, FuncItem) and
+                    isinstance(override.definition, FuncItem)):
+                if original.definition.is_static != override.definition.is_static:
```

Comment:
I'm not sure why staticmethod is treated different from classmethod here -- an instance method can be overridden by a classmethod (if the instance is not needed but the class is) or by a staticmethod (if the subclass implementation needs neither). Or am I missing some cleverly constructed example here? (Totally possible.)

Also it seems that a staticmethod can safely override a classmethod, *and* vice versa. (Again I can't think of a counterexample -- they just differ in whether the class gets passed into the implementation, but the signature of the resulting object (after possibly binding the class) is always the same.

(IIUC, the reason a classmethod cannot be overridden by an instance (default) method is that B.foo and C.foo don't have the same signature if B.foo is a classmethod and C.foo is an instance method -- the latter still requires an instance to be passed in. But this doesn't apply to the other examples.)